ouroboros-network-framework/test/Main.hs

```diff
@@ -0,0 +1,30 @@
+module Main (main) where
+
+import Test.Tasty
+
+import Test.Ouroboros.Network.ConnectionManager qualified as ConnectionManager
+import Test.Ouroboros.Network.Driver qualified as Driver
+import Test.Ouroboros.Network.RateLimiting qualified as RateLimiting
+import Test.Ouroboros.Network.RawBearer qualified as RawBearer
+import Test.Ouroboros.Network.Server2 qualified as Server2
+import Test.Ouroboros.Network.Socket qualified as Socket
+import Test.Ouroboros.Network.Subscription qualified as Subscription
+import Test.Simulation.Network.Snocket qualified as Snocket
+
+main :: IO ()
+main = defaultMain tests
+
+tests :: TestTree
+tests =
+  testGroup "ouroboros-network-framework"
+  [ ConnectionManager.tests
+  , Driver.tests
+  , Server2.tests
+  , Socket.tests
+  , Subscription.tests
+  , RateLimiting.tests
+  , Snocket.tests
+  , RawBearer.tests
+  ]
```

Comment:
I think you want to add the tests to `ouroboros-network-framework:io-tests`; the `Main.hs` might be an artefact of a rebase error.

Reply:
Refactored things into the two test suites.